components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionController.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.webextensions
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.webextension.MessageHandler
+import mozilla.components.concept.engine.webextension.WebExtension
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Provides functionality to feature modules that need to interact with a web extension.
+ *
+ * @property extensionId the unique ID of the web extension e.g. mozacReaderview.
+ * @property extensionUrl the url pointing to a resources path for locating the
+ * extension within the APK file e.g. resource://android/assets/extensions/my_web_ext.
+ */
+class WebExtensionController(
```

Comment:
I'm bit confused if this is a building block why is not part of the `mozilla.components.concept.engine.webextension` we normally use the `support` packages for utilities and extensions.

Reply:
@Amejia481 yeah that's fair. this is because it's not a "concept", it actually has concrete behaviour, but let's work on moving this to the actual engine implementation in the future (see comment above).

Reply:
👍

Reply:
@Amejia481 For implementing https://github.com/mozilla-mobile/android-components/issues/4438 I think we're also gonna need to move the registry of installed extensions to the engine. So, that's a good next step. I just mentioned that to @psymoon as well.

We will likely keep support-webextensions thoughm similar to support-base for reusable building blocks...but yeah let's talk more.

Reply:
Great!